src/runtimes/node/bundlers/esbuild/plugin_in_source_config.ts

```diff
@@ -0,0 +1,21 @@
+import type { Plugin } from '@netlify/esbuild'
+
+import { IN_SOURCE_CONFIG_MODULE } from '../../in_source_config'
+
+// An esbuild plugin that collects the paths of any files that import the
+// in-source configuration module.
+const getISCImportFinderPlugin = ({ importerPaths }: { importerPaths: Set<string> }): Plugin => ({
+  name: 'find-isc-imports',
+  setup(build) {
+    const filter = new RegExp(`^${IN_SOURCE_CONFIG_MODULE}$`)
+
+    build.onResolve({ filter }, (args) => {
+      importerPaths.add(args.importer)
+
+      // eslint-disable-next-line unicorn/no-useless-undefined
+      return undefined
```

Comment:
This is odd, but without this return I'm getting a TypeScript error:

```
Argument of type '(args: OnResolveArgs) => void' is not assignable to parameter of type '(args: OnResolveArgs) => OnResolveResult | Promise<OnResolveResult | null | undefined> | null | undefined'.
  Type 'void' is not assignable to type 'OnResolveResult | Promise<OnResolveResult | null | undefined> | null | undefined'.ts(2345)
```

Reply:
I was also surprised by this at first, but found a good reasoning:

> It's not the same as `undefined` because a function returning `void` might, in javascript at runtime, return something other than `undefined`.
> ~ [Source](https://github.com/Microsoft/TypeScript/issues/30880#issuecomment-482547767)

Reply:
What confuses me is that we _are_ returning `undefined` in this exact same esbuild plugin hook elsewhere in the codebase: https://github.com/netlify/zip-it-and-ship-it/blob/main/src/runtimes/node/bundlers/esbuild/plugin_native_modules.ts#L40. So why does it work in one instance but not the other?

Reply:
Because the other doesn't return `void`, but `Promise<void>`.

Reply:
Same deal if I make this one async.

![Screenshot 2021-11-23 at 16 54 04](https://user-images.githubusercontent.com/4162329/143068796-113c1c48-34ad-4c91-b061-0ce68138a1ea.png)